### PR TITLE
Add better support for release groups in build-tools

### PR DESCRIPTION
**Pull request by tylerbutler**
- Updates several MonoRepoKind uses that were incorrect.
- Formalizes the version schemes into a string union type with the three supported schemes.
- Condenses the existing version translation function into a new adjustVersions function that can support the new schemes.
- Adds a new bump type, "current" that can be used to bump a monorepo to its expected (current) version as defined in lerna.json. This can be useful to clean up versions when merging changes from between branches.

## Reviewer guide

The first three commits in this PR can be reviewed individually. It may be helpful to look at the changes in that way so the number of changes isn't overwhelming.

### CODE REVIEW

File: build-tools/packages/build-tools/src/bumpVersion/bumpVersion.ts [hunk 1]
```diff
@@ -123,29 +123,122 @@ function translateVirtualVersion(
     virtualVersion.format(); // semver must be reformated after edits
     return virtualVersion;
 }
+
+export async function setReleaseGroupVersion(
```
Comment (tylerbutler) [inline build-tools/packages/build-tools/src/bumpVersion/bumpVersion.ts:127]: This function should wait for another PR.

File: build-tools/packages/build-tools/src/bumpVersion/bumpVersion.ts [hunk 2]
```diff
@@ -123,29 +123,122 @@ function translateVirtualVersion(
...
     for (const monoRepo of monoReposNeedBump) {
-        console.log(`  Bumping ${monoRepo} version${vPatchLogString}`);
+        console.log(`  Bumping ${monoRepo}${vPatchLogString}...`);
         // Translate the versionBump into the appropriate change for virtual patch versioning
-        const translatedVersionBump = translateVirtualVersion(versionBump, versionBag.get(monoRepo), virtualPatch);
+        const ver = getVersion(monoRepo);
+        assert(ver, "ver is missing");
+        const adjVer = await adjustVersion(ver, versionBump);
+        const translatedVersionBump = adjVer!;
+        // translateVirtualVersion(adjVer!, versionBag.get(monoRepo), virtualPatch);
```
Comment (tylerbutler) [inline build-tools/packages/build-tools/src/bumpVersion/bumpVersion.ts:241]: We still use the virtual patch format in some places, so probably need to keep this call or move it to adjustVersion.

File: build-tools/packages/build-tools/src/bumpVersion/context.ts [hunk 3]
```diff
@@ -179,4 +185,10 @@ export class Context {
             await this.gitRepo.deleteTag(tag);
         }
     }
+
+    public packagesForReleaseGroup(releaseGroup: MonoRepoKind) {
```
Comment (tylerbutler) [inline build-tools/packages/build-tools/src/bumpVersion/context.ts:189]: add docs

File: build-tools/packages/build-tools/src/bumpVersion/utils.ts [hunk 4]
```diff
@@ -46,3 +48,111 @@ export function prereleaseSatisfies(packageVersion: string, range: string) {
...
+                    return sv?.inc(bumpType) ?? null;
+                default:
+                    // If the bump type is an explicit version, just use it.
+                    return bumpType;
+            }
+        }
+        case "internal": {
+            fatal("Not yet implemented");
+            break;
+        }
```
Comment (tylerbutler) [inline build-tools/packages/build-tools/src/bumpVersion/utils.ts:144]: Future PR.

File: build-tools/packages/build-tools/src/common/fluidRepo.ts [hunk 5]
```diff
@@ -67,23 +67,15 @@ export class FluidRepo {
             return Packages.loadDir(item.directory, group, item.ignoredDirs);
         }
 
+        // const releaseGroups = new Set([...supportedMonoRepoValues()].map(g => g.toString()));
+
```
Comment (tylerbutler) [inline build-tools/packages/build-tools/src/common/fluidRepo.ts:71]: remove

File: build-tools/packages/build-tools/src/bumpVersion/utils.ts [hunk 6]
```diff
@@ -46,3 +48,111 @@ export function prereleaseSatisfies(packageVersion: string, range: string) {
...
+ * "semver" is the standard semver scheme.
+ *
+ * "internal" is the 2.0.0-internal.1.0.0 scheme.
+ *
+ * "virtualPatch" is the 0.36.1002 scheme.
+ */
+export type VersionScheme = "semver" | "internal" | "virtualPatch";
+
+/**
+ *
```
Comment (Josmithr) [inline build-tools/packages/build-tools/src/bumpVersion/utils.ts:115]: Summary docs?

File: build-tools/packages/build-tools/src/bumpVersion/utils.ts [hunk 7]
```diff
@@ -46,3 +48,111 @@ export function prereleaseSatisfies(packageVersion: string, range: string) {
     // Pretend that the current package is latest prerelease (zzz) and see if the version still satisfies.
     return semver.satisfies(`${packageVersion}-zzz`, range)
 }
+
+/**
+ * Translate a VersionChangeType for the virtual patch scenario where we overload a beta version number
```
Comment (Josmithr) [inline build-tools/packages/build-tools/src/bumpVersion/utils.ts:53]: Nit
```suggestion
 * Translate a {@link VersionChangeType} for the virtual patch scenario where we overload a beta version number
```

At least in theory should help prevent doc staleness if we use VSCode refactoring tools

File: build-tools/packages/build-tools/src/bumpVersion/context.ts [hunk 8]
```diff
@@ -179,4 +185,16 @@ export class Context {
             await this.gitRepo.deleteTag(tag);
         }
     }
+
+    /**
+     * Returns the packages that belong to the specified release group.
+     *
+     * @param releaseGroup The release group to filter by
```
Comment (Josmithr) [inline build-tools/packages/build-tools/src/bumpVersion/context.ts:192]:
```suggestion
     * @param releaseGroup - The release group to filter by
```

**Review by Josmithr — APPROVED** [review]

File: build-tools/packages/build-tools/src/bumpVersion/utils.ts (hunk 6 above)
Comment (tylerbutler) [reply build-tools/packages/build-tools/src/bumpVersion/utils.ts:116]: Added.

**Review by Josmithr — APPROVED** [review]

**Pull request merged**